activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -732,11 +732,31 @@ def raw_execute(sql, name, async: false, allow_retry: false, uses_transaction: t
           log(sql, name, async: async) do
             with_raw_connection(allow_retry: allow_retry, uses_transaction: uses_transaction) do |conn|
               sync_timezone_changes(conn)
-              conn.query(sql)
+              result = conn.query(sql)
+              handle_warnings(sql)
+              result
             end
           end
         end
 
+        def handle_warnings(sql)
+          return if @raw_connection.warning_count == 0
+
+          result = @raw_connection.query("SHOW WARNINGS")
+          result.each do |level, code, message|
+            warning = SQLWarning.new(message, code, level, sql)
+            next if warning_ignored?(warning)
+
+            ActiveRecord.db_warnings_action.call(warning)
+          end
+        end
+
+        def warning_ignored?(warning)
+          return true if warning.level == "Note"
+
+          ActiveRecord.db_warnings_ignore.any? { |warning_matcher| warning.message.match?(warning_matcher) }
```

Comment:
This last line seems common across adapters. Should we extract it to the abstract adapter and call super here and at the end of the similar definition in postgresql?